### PR TITLE
Android workmanager

### LINKED ISSUE #662: Getting started with WorkManager
### Brief Summary: 
Performing background tasks in Android has always been a challenge. Jetpack introduced the WorkManager library to make this process easier for developers. This article aims at giving the reader a basic understanding of what the library is and it's pros.

### Key Takeaways: 
1. What is WorkManager
2. How to use WorkManager
3. The advantages of using WorkManager

### References: 
N/A

### PR DESCRIPTION
**Pull request by LinusMuema**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

this fixes #662

### CODE REVIEW

File: articles/android-workmanager/index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+Now that we have our work completely set up. You can go ahead and follow through the repository to check the UI setup.
+
+Once the application starts, it calls the `startWork` function which starts our work. This starts both the `oneTimeWork` and `periodicWork`. The one time work completes immediately and adds one user to the Room database. The `periodicWork` also starts by adding one user to the database and another user after 15 minutes. So after 2 hours, the `periodicWork` will have added 8 users. On the first run the application should show 2 users. One from the `oneTimeWork` and another from the `periodicWork`.
+
+### Conclusion
+With that, you have the basic information about `WorkManager`. As you can see, it is easier to schedule background tasks. You are able to run tasks like network calls even if the application closes. A good use case could be backing up data on a different network like cloud services. This can be done when some constraints are met and you have the assurance of your work being done. Go ahead and clone the [repo](https://github.com/LinusMuema/kotlin/tree/workManager) and run the application. Feel free to raise a PR or an issue with any updates.
```
Comment (kayere) [inline articles/android-workmanager/index.md:162]: I'd suggest you let the reader download the repo at the beginning (first step). Then you walk the reader through adding the different classes and methods associated with the technology. I think that would be better.

**Review by kayere — COMMENTED** [review]
Hello @LinusMuema. Great article. I was going through the tech review of your article and I got a suggestion.  Please check it in the inline comment.

File: articles/android-workmanager/index.md (hunk 1 above)
Comment (LinusMuema) [reply articles/android-workmanager/index.md:162]: On it...

**Comment by LinusMuema** [comment]
Hey @kayere 
I have made the suggestions...

**Review by kayere — APPROVED** [review]

**Pull request merged**